onnx2tf/onnx2tf.py

```diff
@@ -1963,18 +1966,30 @@ def main():
     #   [{input_op_name} {numpy_file_path} {mean} {std}],
     #   [{input_op_name} {numpy_file_path} {mean} {std}],
     # ]
-    calib_params = []
-    if args.quant_calib_input_op_name_np_data_path is not None:
-        for param in args.quant_calib_input_op_name_np_data_path:
-            input_op_name = str(param[0])
-            numpy_file_path = str(param[1])
-            mean = np.asarray(ast.literal_eval(param[2]), dtype=np.float32)
-            std = np.asarray(ast.literal_eval(param[3]), dtype=np.float32)
-            calib_params.append(
-                [input_op_name, numpy_file_path, mean, std]
-            )
-    if len(calib_params) == 0:
-        calib_params = None
+    custom_params = []
+    if args.custom_input_op_name_np_data_path is not None:
+        for param in args.custom_input_op_name_np_data_path:
+            tmp = []
+            if len(param) == 2:
+                tmp.append(str(param[0])) # input_op_name
+                tmp.append(str(param[1])) # numpy_file_path
+                
+                if len(param) == 4:
+                    tmp.append(np.asarray(ast.literal_eval(param[2]), dtype=np.float32)) # mean
+                    tmp.append(np.asarray(ast.literal_eval(param[3]), dtype=np.float32)) # std
+                    
+                custom_params.append(
+                    tmp
+                )
+            else:
+                error_msg = f'' + \
+                            f'{Color.RED}ERROR:{Color.RESET} ' + \
+                            f"'-cind' option must have INPUT_NAME, NUMPY_FILE_PATH, MEAN(optional), STD(optional)"
+                print(error_msg)
+                raise ValueError(error_msg)
+            
+    if len(custom_params) == 0:
+        custom_params = None
```

Comment:
You are correct that they implemented input checks. However, it is not an appropriate place to implement. This tool can be imported and used as a Python script. Therefore, in order to efficiently perform the same input check when executed from a Python script, it is better to implement the error determination in the range around lines 424 to 540. In the range of lines 1969 to 1992, it is cleaner to take on only the role of processing parameters.

Reply:
- **[lines 425 to 433]**
```python
    # determination of errors in custom input
    if custom_input_op_name_np_data_path is not None:
        for param in custom_input_op_name_np_data_path:
            if len(param) not in [2, 4]:
                error_msg = f'' + \
                            f'{Color.RED}ERROR:{Color.RESET} ' + \
                            f"'-cind' option must have INPUT_NAME, NUMPY_FILE_PATH, MEAN(optional), STD(optional)"
                print(error_msg)
                raise ValueError(error_msg)
```

- **[lines 1979 to 1936]**

```python
    custom_params = []
    if args.custom_input_op_name_np_data_path is not None:
        for param in args.custom_input_op_name_np_data_path:
            tmp = []
            if len(param) == 2:
                tmp.append(str(param[0])) # input_op_name
                tmp.append(str(param[1])) # numpy_file_path
                
                if len(param) == 4:
                    tmp.append(np.asarray(ast.literal_eval(param[2]), dtype=np.float32)) # mean
                    tmp.append(np.asarray(ast.literal_eval(param[3]), dtype=np.float32)) # std
                    
                custom_params.append(
                    tmp
                )

    if len(custom_params) == 0:
        custom_params = None
```

Would this implementation be appropriate?

Reply:
You seem to have understood my intent. Thank you. Looks good. :+1: